spring-boot-3-integrationtests/src/test/java/org/axonframework/springboot/connection/SpringBootDockerComposeIntegrationTest.java

```diff
@@ -0,0 +1,51 @@
+package org.axonframework.springboot.connection;
+
+import io.axoniq.axonserver.connector.AxonServerConnection;
+import io.axoniq.axonserver.connector.AxonServerConnectionFactory;
+import org.axonframework.axonserver.connector.AxonServerConfiguration;
+import org.axonframework.axonserver.connector.AxonServerConnectionManager;
+import org.axonframework.utils.AssertUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class SpringBootDockerComposeIntegrationTest {
+
+    private ConfigurableApplicationContext application;
+
+    @BeforeEach
+    void setUp() {
+        application = SpringApplication.run(SpringBootApplication.class,
+                                            "--spring.docker.compose.file=test-docker-compose.yml",
+                                            "--spring.docker.compose.skip.in-tests=false");
+    }
+
+    @AfterEach
+    void tearDown() {
+        application.stop();
+    }
+
+    @Test
+    void verifyApplicationRunsAndConnectsToAxonServerDefinedInDockerComposeFile() {
+        assertTrue(application.isRunning());
+        AxonServerConfiguration config = application.getBean(AxonServerConfiguration.class);
+
+        assertNotNull(application.getBean(AxonServerConnectionDetails.class), "Expected an AxonServerConnectionDetails bean pointing to Axon Server in Docker");
+
+        assertNotNull(config);
+        assertNotEquals("localhost:8124", config.getServers());
+        assertNotEquals("localhost", config.getServers());
+
+        AxonServerConnectionManager connectionFactory = application.getBean(AxonServerConnectionManager.class);
+        AxonServerConnection connection = connectionFactory.getConnection();
+        AssertUtils.assertWithin(5, TimeUnit.SECONDS, () -> assertTrue(connection.isConnected()));
```

Comment:
We've been moving a lot of our custom `AssertUtils` to "awaitility" for it's increased flexibility.
Not that our `assertWithin` doesn't work anymore, just wanted to raise awareness on the subject of time-sensitive tests.